### ISSUE TITLE
feature request: guidance/docs on how to use experimental-utils with astexplorer.net

### ISSUE THREAD

**Issue by benmonro** [typescript-eslint/typescript-eslint#2880]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
This is not a bug, but rather a feature request for docs on how to use ESLintUtils.RuleCreator with astexplorer, or if another alternative exists what that is.

```TS
// your repro code case
```

**Expected Result**

**Actual Result**

**Additional Info**

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `X.Y.Z` |
| `TypeScript`                            | `X.Y.Z` |
| `node`                                  | `X.Y.Z` |

**Comment by bradzacher** [comment]
What do you mean by "how to use ESLintUtils.RuleCreator with astexplorer"

What are you looking to do exactly?

**Comment by benmonro** [comment]
well, specifically, I use https://astexplorer.net/ during the development process for new lint rules.  I've been using this for years w/ javascript.  I'm a bit new to writing lint rules in typescript, but I was hoping to still be able to use that tool as it provides a great dev experience for exploring the AST etc.

so for example I have this function: 

```
export const createRule = ESLintUtils.RuleCreator(name => {

    const REPO_URL = "https://github.com/testcafe-community/eslint-plugin-testcafe-community";
    const ruleName = parsePath(name).name;
    return `${REPO_URL}/blob/v${version}/docs/rules/${ruleName}.md`;
  });
```

which my rule then uses: 
```

export default createRule({
    name: __filename,
    defaultOptions: [],
    meta: {
        type:"problem",
        messages: {
            noIdenticalTitles: "Don't use identical titles for your tests"
        },
        docs: {
            description: "Don't use identical titles for your tests",
            category: "Best Practices",
            recommended: "error"
        },
        schema: []
    },
    
    create(context) { ... }
 });
```

so I'm just curious what (if anything) you guys recommend for using either astexplorer or some other tooling to provide that experience in conjunction w/ this lib...

**Comment by bradzacher** [comment]
unfortunately there's no way make it work.
ASTExplorer does its own parsing and evaluation of the text you enter, so it doesn't support things like imports.

I personally don't use ASTExplorer for writing rules beyond _really_ quick-and-dirty prototypes. There are various problems with it namely when you turn on "transform" mode, it locks the parser to `babel-eslint`, preventing you from testing with our parser.

My workflow work writing lint rules is to run `jest --watch <test>`, and I have a test file which looks like
```ts
// imports and preamble...

ruleTester.run('foo', rule, {
  valid: [
    `
case i want to dev against
    `,
  ],
  invalid: [],
});
```

That way I get the safety of TS, and the tools of my IDE (intellisense, etc), whilst still having live feedback.
I also have astexplorer open separately in case I need to interrogate the AST, but ofc it's not in transform mode, as I don't need it.

**Comment by benmonro** [comment]
hmm, that's a bummer, I'm a big fan of using ast explorer.

**Comment by bradzacher** [comment]
It's all open source!
If you want to enable the tooling in astexplorer - support is just a PR away - https://github.com/fkling/astexplorer
